client/src/index.js

```diff
@@ -40,23 +44,68 @@ const IEMessage = () => {
 
 const authIssuer = process.env.REACT_APP_OUTPUT === 'ALL' ? AuthIssuer.GOA : AuthIssuer.User;
 
-const App = () => (
-  <ThemeProvider theme={Theme}>
-    <BrowserRouter>
-      <ToastProvider>
-        <AuthProvider authIssuer={authIssuer} userType={AuthIssuer.User}>
-          <ModalProvider>
-            <ErrorBoundary>
-              <CssBaseline />
-              <Toast />
-              <Modal />
-              <AppRoutes />
-            </ErrorBoundary>{' '}
-          </ModalProvider>{' '}
-        </AuthProvider>{' '}
-      </ToastProvider>{' '}
-    </BrowserRouter>{' '}
-  </ThemeProvider>
-);
+export const AuthContext = React.createContext({
+  tokensInitialized: false,
+});
+
+const App = () => {
+  const [authState, setAuthState] = useState({
+    tokensInitialized: false,
+  });
+
+  const handleTokens = (tokens) => {
+    AxiosPrivate.defaults.headers.common['Authorization'] = `Bearer ${tokens.token}`;
+    setAuthState((prev) => ({ ...prev, tokensInitialized: true }));
+  };
+
+  return (
+    <ReactKeycloakProvider
+      LoadingComponent={
+        <Box
+          height='100vh'
+          width='100%'
+          display='flex'
+          justifyContent='center'
+          flexDirection='column'
+          alignItems='center'
+        >
+          <CircularProgress />
+          <Typography
+            style={{
+              fontFamily: 'acumin-pro-semi-condensed, sans-serif',
+              fontWeight: '600',
+              fontSize: '1.5rem',
+              paddingTop: '1rem',
+            }}
+            variant='h1'
+          >
+            Authenticating...
+          </Typography>
+        </Box>
+      }
+      authClient={keycloakClient}
+      onTokens={handleTokens}
+    >
+      <AuthContext.Provider value={authState}>
+        <ThemeProvider theme={Theme}>
+          <BrowserRouter>
+            <ToastProvider>
+              <AuthProvider authIssuer={authIssuer} userType={AuthIssuer.User}>
```

Comment:
We can remove this provider as we've Keycloak provider now. But can be done as a part of moving the code

Reply:
Good point, I'll make a note of that